wordpress-mobile/WordPress-iOS
# #8950: Update WordPressKit

First part of #8950 

This PR includes some additions related to _WordPressKit_.
I had to create 2 new models to map the site delivery methods and add the relationships with `RemoteReaderSiteInfo`.
I updated the API version within the current `ReaderTopicRemoteService` and i created 3 protocols to handle the new actions (in this case i can use the same subset for the local service as well). Also i extended the remote service (using Swift) implementing these new protocols to handle the new actions.
I even added a NSObject category to check if an object is valid and not nil. This category it's used by the remote service and the local service to check if an object is valid.
Unit Test are implemented to test the Remote Service and the new models

**To test:**
The changes are affecting only the WordPressKit. There are different Uni Tests to run and check the new features. Also i think a sanity test might be useful.
